src/conventional/sonde_tac2ioda.py

```diff
@@ -966,8 +970,12 @@ def change_vars(profile):
         if 'height' in levels[pressure] and levels[pressure]['height'] is not None:
             height = levels[pressure]['height']
             dz = height - heightKm1
-            # Typical radiosonde ascent rate is 5 m/s
-            this_datetime = previous_time + timedelta(seconds=dz*0.2)
+            # Legacy soundings produce null values at mandatory level below ground.
+            if (dz < 1.0):
+                this_datetime = previous_time
+            else:
+                # Typical radiosonde ascent rate is 5 m/s
+                this_datetime = previous_time + timedelta(seconds=dz*0.2)
```

Comment:
Dividing by `5.0` here might make the intent clearer. Up to you.

Reply:
Yes, it would be more clear and yet I often multiply by the inverse number for making code run as fast as possible since sometimes division is slower than multiplication and the comment above helps make it clear.